### PR TITLE
Trying to create a directory in a file should return ENOTDIR not EPERM

### PR DESCRIPTION
**Pull request by hoodmane**
In linux, `mkdir("some_file/dir")` returns `ENOTDIR` but Emscripten returns `EPERM`. This fixes it.

### CODE REVIEW

File: src/library_memfs.js [hunk 1]
```diff
@@ -7,7 +7,7 @@
 addToLibrary({
   $MEMFS__deps: ['$FS', '$mmapAlloc'],
 #if !ASSERTIONS
-  $MEMFS__postset: `
+  $MEMFS__preset: `
```
Comment (hoodmane) [inline src/library_memfs.js:10]: We sometimes thrown `MEMFS.doesNotExistError` inside `FS.staticInit()` which happens before we get to `MEMFS__postset`.

File: src/library_fs.js [hunk 2]
```diff
@@ -376,6 +376,9 @@ FS.staticInit();
         var node = FS.lookupNode(dir, name);
         return {{{ cDefs.EEXIST }}};
       } catch (e) {
+        if (e.errno === {{{ cDefs.ENOTDIR }}}) {
+          return {{{ cDefs.ENOTDIR }}};
+        }
```
Comment (hoodmane) [inline src/library_fs.js:381]: Maybe it would be better to check for this before calling `mayCreate` though...

File: test/test_core.py [hunk 3]
```diff
@@ -5822,6 +5822,10 @@ def test_fs_write(self):
   def test_fs_emptyPath(self):
     self.do_run_in_out_file_test('fs/test_emptyPath.c')
 
+  @also_with_noderawfs
+  def test_enotdir(self):
```
Comment (sbc100) [inline test/test_core.py:5826]: test_enotdir -> test_fs_enotdir

**Review by sbc100 — APPROVED** [review]

File: test/fs/test_enotdir.c [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+int main() {
+    {
+        int src_fd = open("file", O_CREAT | O_WRONLY, 0777);
+        close(src_fd);
+    }
+    {
+        int target_fd = mkdir("file/blah", 0777);
+        printf("target_fd: %d, errno: %d %s\n", target_fd, errno, strerror(errno));
+    }
```
Comment (sbc100) [inline test/fs/test_enotdir.c:16]: 2-space indentation.

**Review by sbc100 — APPROVED** [review]

**Comment by hoodmane** [comment]
@sbc100 looks like `test_nodejs_sockets_echo_subprotocol` failed with `Exception: [Websockify failed to start up in a timely manner]` twice in a row...

**Comment by hoodmane** [comment]
It's also failing on #23017.

**Comment by sbc100** [comment]
> It's also failing on #23017 and #22925.

yes it seems to be very flaky recently.  I'm not sure why

File: test/test_core.py [hunk 5]
```diff
@@ -5832,6 +5832,10 @@ def test_fs_write(self):
   def test_fs_emptyPath(self):
     self.do_run_in_out_file_test('fs/test_emptyPath.c')
 
+  @also_with_noderawfs
```
Comment (sbc100) [inline test/test_core.py:5835]: Oh I forget to say we should me marking these as @crossplatform so that they run on windows.

**Pull request merged**